src/actions/auth.js

```diff
@@ -0,0 +1,246 @@
+const StrangeAuth = require('strange-auth');
+const WebClient = require('../utils/web-client');
+const History = require('../wiring/history');
+const AuthAct = require('../action-types/auth');
+
+const internals = {};
+
+const actions = exports;
+
+// New User Registration
+
+exports.registrationRequest = () => {
+
+    return {
+        type: AuthAct.REGISTRATION_BEGIN
+    };
+};
+
+exports.registrationSuccess = () => {
+
+    return {
+        type: AuthAct.REGISTRATION_SUCCESS
+    };
+};
+
+exports.registrationFailure = (errMessage) => {
+
+    return {
+        type: AuthAct.REGISTRATION_FAILURE,
+        payload: errMessage
+    };
+};
+
+exports.registerUser = ({ email, password, firstName, lastName }) => {
+
+    return (dispatch) => {
+
+        dispatch(actions.registrationRequest());
```

Comment:
Doesn't matter so much for this PR, but I like passing the action payload through here for any reducers that may be interested.